fetch.bs

```diff
@@ -2050,6 +2050,34 @@ clearly stipulates that <a>connections</a> are keyed on
      WebSocket saner -->
 
 
+<h3 id=http-cache-partitions>HTTP cache partitions</h3>
+
+<p>To
+<dfn lt="determine the HTTP cache partition|determining the HTTP cache partition">determine the HTTP cache partition</dfn>,
+given <var>request</var>, run these steps:
+
+<ol>
+ <li><p>If <var>request</var> is a <a>navigation request</a>, set <var>environmentSettings</var> to
+ <var>request</var>'s <a for="environment">target browsing context</a>'s <a>active document</a>'s
+ <a>environment settings object</a>. Otherwise let <var>environmentSettings</var> be
+ <var>request</var>'s <a for=request>client</a>.
```

Comment:
You need the request's reserved client before you can get to a target browsing context. Any reason not to use the creation URL's origin of the reserved client?

Reply:
Added the request's reserved client.
Regarding using the creation URL's origin, in case of iframe, that will not give the top-level origin, right and we want the top-level origin as the partitioning key for an iframe navigation request

Reply:
That's a good point, but I don't think this works either as the active document isn't the document we are navigating towards so it might have the wrong set of origins.

Reply:
We probably need to put top-level origin on an environment.

Reply:
And that's because the active document is not yet set and will be set after this fetch, right? 

Then another option could be to only special case the main frame's navigation fetch request and not the iframe's, since iframe navigation request's partitioning key is the same as that of a subresource in the main frame.
And if we are only special casing the main frame navigation request then we could use the creation URL's origin, I guess. Does that make sense?

Reply:
I'm afraid there's another corner case. If you have A which embeds B and popups C which embeds D, A or B can sometimes navigate D, but we should use the key from C. (Unfortunately it's not clearly defined when A/B can navigate D. It's a very longstanding issue against HTML.)

Reply:
In the scenario that you mention D is an iframe, and if we leave the iframe's navigation request case handled similarly as a subresource, then it should be ok?

Reply:
Here the client would be A/B, not C. For a subresource of C the client would always be C. Frames are different.

Reply:
Got it. 
I was thinking the following for getting the top-level origin for iframe and top-level navigation requests:

- iframe navigation request: request's reserved client's target browsing context's top-level browsing context's active document's origin. As I understand it, In the case given above target browsing context's top-level browsing context will be C and not A/B, right?
- top-level navigation request: request's origin

Reply:
That might work, yeah! It seems cleaner to make top-level origin available on reserved client though.

Reply:
Created https://github.com/whatwg/html/pull/5491 to define the target top-level origin in the reserved client.